src/main/java/com/github/princesslana/eriscasper/rx/websocket/RxWebSocket.java

```diff
@@ -54,6 +66,12 @@ public void onClosed(WebSocket ws, int code, String reason) {
 
     @Override
     public void onClosing(WebSocket ws, int code, String reason) {
+      if (code == 4004) {
+        socket.close(1002, "Invalid token.");
+        em.onNext(ClosingTuple.of(ws, code, reason));
+        throw new ErisCasperFatalException(
+            "Failed to authenticate with discord servers: [" + reason + "]");
```

Comment:
The websocket class shouldn't know about the existence of discord.

Why can't the discord listeners listen for the Closing event and handle this?

Reply:
Didn't even realize this was still in the codebase. I'll move it.